invenio_rdm_records/records/dumpers.py

```diff
@@ -59,3 +65,46 @@ def load(self, data, record_cls):
         # `None` covers the cases where exceptions were raised in _dump
         parent_data.pop(f"{self.key}_start", None)
         parent_data.pop(f"{self.key}_end", None)
+
+
+class LocationsDumper(ElasticsearchDumperExt):
+    """ElasticSearch dumper for location data.
+
+    This does a few things:
+
+    * Removes/adds FeatureCollection and Feature types, as those are
+      constant
+    * Calculates a centroid for each location if a geometry is provided
+
+    Centroid calculation requires shapely, except for the simple case where
+    the geometry is a point.
+    """
+
```

Comment:
moderate: I know we only have one "location" field, but is it worth to have the field `key` configurable as in the EDTF dumper? Not sure if this will ever be the case.

Reply:
I should defer to @lnielsen or @slint on this one.

Reply:
In this case, it would probably just leave it as is.

Reply:
:+1: